component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java

```diff
@@ -0,0 +1,551 @@
+        authenticationContext1.setProperty("getPropertiesFromLocal", null);
+
+        return new Object[][] {
+                {authenticationContext1},
+                {authenticationContext2},
+        };
+    }
+
+    @Test(dataProvider = "backupCodeLoginPageData")
+    public void testGetBackupCodeErrorPage(String tenantDomain, boolean isTenantQualifiedURL) throws
+            AuthenticationFailedException, URLBuilderException {
+        mockStatic(IdentityTenantUtil.class);
+        mockStatic(FileBasedConfigurationBuilder.class);
+        mockStatic(ConfigurationFacade.class);
+        mockStatic(ServiceURLBuilder.class);
+        AuthenticationContext authenticationContext = new AuthenticationContext();
+        authenticationContext.setTenantDomain(tenantDomain);
+        authenticationContext.setProperty(BackupCodeAuthenticatorConstants.BACKUP_CODE_AUTHENTICATION_ERROR_PAGE_URL,
+                "totpauthenticationendpoint/custom/error.jsp");
+        when(FileBasedConfigurationBuilder.getInstance()).thenReturn(fileBasedConfigurationBuilder);
+        when(fileBasedConfigurationBuilder.getAuthenticatorBean(BACKUP_CODE_AUTHENTICATOR_NAME)).
+                thenReturn(authenticatorConfig);
+        when(IdentityTenantUtil.isTenantQualifiedUrlsEnabled()).thenReturn(isTenantQualifiedURL);
+        when(ConfigurationFacade.getInstance()).thenReturn(configurationFacade);
+        when(configurationFacade.getAuthenticationEndpointURL()).thenReturn(BACKUP_CODE_LOGIN_PAGE);
+        when(ServiceURLBuilder.create()).thenReturn(serviceURLBuilder);
+        when(serviceURLBuilder.addPath(anyString())).thenReturn(serviceURLBuilder);
+        when(serviceURLBuilder.build()).thenReturn(serviceURL);
+        when(serviceURL.getAbsolutePublicURL()).thenReturn("testURL");
+        String result = BackupCodeUtil.getBackupCodeErrorPage(authenticationContext);
+        assertNotNull(result);
+    }
+
+    @Test(description = "Test case for getErrorPageFromXMLFile(): getErrorPage from registry file.")
+    public void testGetErrorPageFromXMLFile() throws Exception {
+
+        AuthenticationContext authenticationContext = new AuthenticationContext();
+        authenticationContext.setTenantDomain("wso2.org");
+        authenticationContext.setProperty(BackupCodeAuthenticatorConstants.BACKUP_CODE_AUTHENTICATION_ERROR_PAGE_URL,
+                "totpauthenticationendpoint/custom/error.jsp");
+        assertEquals(backupCodeUtil.getErrorPageFromXMLFile(authenticationContext),
+                "totpauthenticationendpoint/custom/error.jsp");
+    }
+
+    @Test(dataProvider = "localUserData")
+    public void testIsLocalUser(Object authenticationContext, boolean expectedVal) {
+
+        boolean result = backupCodeUtil.isLocalUser((AuthenticationContext) authenticationContext);
+        assertEquals(expectedVal, result);
+
+    }
+
+    @DataProvider(name = "localUserData")
+    public Object[][] datForLocalUser() {
+
+//        test case #1
```

Comment:
Is this intentional? If so format properly.

Reply:
Fix other places as well

Reply:
addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622)